### PR TITLE
feat: gains for user

### LINKED ISSUE #2822: Gains for user
Same as we have flippers at collection we would likely replicate it for the user. 
Goal would be to see gains per collection on the user profile. 

Headers would be 
- collection name, amounts, bought, sold, percentage (same as flippers), last activity
- on click user can see items from collection

Hence @Jarsen136 has here experience 
with grapql would like to do this one?

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #2822
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [x] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [x] I've tested PR on mobile and everything seems works
- [x] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
image

### CODE REVIEW

**Comment by yangwao** [comment]
user smtan0x, seems at gains had three items, two of them sold, but on collection in flipper chart is only one item under his name?

https://deploy-preview-2945--koda-nuxt.netlify.app/rmrk/collection/dac5c7f54029d0e73c-HXAI8?tab=flippers

image

vs

https://deploy-preview-2945--koda-nuxt.netlify.app/rmrk/u/DSKK9i2KdyyJ1xrDkcuLSNacLFAhq23TJkDsxtNDvKKR2qY?tab=gains

image

**Comment by Jarsen136** [comment]
> user smtan0x, seems at gains had three items, two of them sold, but on collection in flipper chart is only one item under his name?
> 
> https://deploy-preview-2945--koda-nuxt.netlify.app/rmrk/collection/dac5c7f54029d0e73c-HXAI8?tab=flippers
> 
> image
> 
> vs
> 
> https://deploy-preview-2945--koda-nuxt.netlify.app/rmrk/u/DSKK9i2KdyyJ1xrDkcuLSNacLFAhq23TJkDsxtNDvKKR2qY?tab=gains
> 
> image


This display result is correct. @yangwao 

flipper: show the total current owners of a collection
gains: show the trade history of a user


as for user smtan0x, he only owns one nft for now. 
image

**Comment by Jarsen136** [comment]
Could you pls give me some more code review comments? @vikiival @roiLeo   👀

**Comment by Jarsen136** [comment]
> user smtan0x, seems at gains had three items, two of them sold, but on collection in flipper chart is only one item under his name?
> 
> https://deploy-preview-2945--koda-nuxt.netlify.app/rmrk/collection/dac5c7f54029d0e73c-HXAI8?tab=flippers
> 
> image
> 
> vs
> 
> https://deploy-preview-2945--koda-nuxt.netlify.app/rmrk/u/DSKK9i2KdyyJ1xrDkcuLSNacLFAhq23TJkDsxtNDvKKR2qY?tab=gains
> 
> image

Already fixed.
Now it would hide burned NFT in the gains tab.

image

**Comment by yangwao** [comment]
Oh actually. Gains and Flipper should show even burned NFTs bc gain has happened for the user thanks to trade, their balance didn't been wiped out 😛

And hence some burns happen bc of migration to rmrkv2, I guess we should preserve it for historical value and insights :)

**Comment by Jarsen136** [comment]
> Oh actually. Gains and Flipper should show even burned NFTs bc gain has happened for the user thanks to trade, their balance didn't been wiped out 😛
> 
> And hence some burns happen bc of migration to rmrkv2, I guess we should preserve it for historical value and insights :)

Oh, I see. I will try to make it.

**Comment by Jarsen136** [comment]
Changelog:

- [x] 1. fix: Gains and Flipper should show even burned nft.  
image

- [x]  2. fix: History on collection should show burn event.   
image

**Review by vikiival — COMMENTED** [review]
oof such big query :)

File: queries/rmrk/subsquid/allCollectionSaleEvents.graphql [hunk 1]
```diff
@@ -1,10 +1,11 @@
 query allCollectionSaleEvents($id: ID!, $and: [EventWhereInput!]) {
-  events(where: { nft: {burned_eq: false, collection: { id_eq: $id } }, AND: $and }) {
```
Comment (roiLeo) [inline queries/rmrk/subsquid/allCollectionSaleEvents.graphql:2]: why did you remove this from query?

File: queries/rmrk/subsquid/allNftSaleEventsHistoryByAccountId.graphql [hunk 2]
```diff
@@ -0,0 +1,21 @@
+query allNftSaleEventsHistoryByAccountId($id: String) {
+  events(
+    where: { OR: [{ currentOwner_eq: $id }, { caller_eq: $id }, { meta_eq: $id }] }) {
```
Comment (roiLeo) [inline queries/rmrk/subsquid/allNftSaleEventsHistoryByAccountId.graphql:3]: I would add burned filter here too

**Review by roiLeo — COMMENTED** [review]
Is it not possible to use fragments in your queries?

**Comment by Jarsen136** [comment]
> Is it not possible to use fragments in your queries?

Yes, I have add an event fragment.

**Comment by yangwao** [comment]
I guess good to go! 
@Jarsen136 last few things to do before merge
- when I'm clicking from Flippers on username it should link to gains tab 
image
- resolve conflicts and let's merge it! :)

**Comment by yangwao** [comment]
pay 150 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $150 @ 69.65 USD/KSM ~ 2.154 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0xacd0ff1bcecf0925f183aa66788fac4ace50d583d5b459d1f971bf1a282cd551](https://kusama.subscan.io/extrinsic/0xacd0ff1bcecf0925f183aa66788fac4ace50d583d5b459d1f971bf1a282cd551)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by roiLeo** [comment]
😕 what about burned nfts?

**Comment by Jarsen136** [comment]
> 😕 what about burned nfts?

Seems that you are confused if it should show the burned nft info at Gains and Flipper. : )
Maybe @yangwao could answer this. 

#ref https://github.com/kodadot/nft-gallery/pull/2945#issuecomment-1120241022
#ref https://github.com/kodadot/nft-gallery/pull/2945#issuecomment-1120249012

**Comment by roiLeo** [comment]
> > 😕 what about burned nfts?
> 
> Seems that you are confused if it should show the burned nft info at Gains and Flipper. : ) Maybe @yangwao could answer this.
> 
> #ref [#2945 (comment)](https://github.com/kodadot/nft-gallery/pull/2945#issuecomment-1120241022) #ref [#2945 (comment)](https://github.com/kodadot/nft-gallery/pull/2945#issuecomment-1120249012)

Okay, thanks for the answer. I was afraid that removing burnt nft would distort statistics.
